master/internal/rm/kubernetesrm/pods.go

```diff
@@ -934,99 +937,92 @@ func (p *pods) handleGetAgentsRequest(ctx *actor.Context) {
+
+		for poolName, tcd := range poolTaskContainerDefaults {
+			var poolTolerations []k8sV1.Toleration
+
+			// If they're using the default RP config, use the default tolerations.
+			if len(p.resourcePoolConfigs) <= 1 &&
+				(tcd == nil || (tcd.CPUPodSpec == nil && tcd.GPUPodSpec == nil)) {
+				poolTolerations = defaultTolerations
+			} else if tcd != nil {
+				// Decide which poolTolerations to use based on slot device type
+				if slotType == device.CUDA && tcd.GPUPodSpec != nil {
+					//nolint:gocritic,appendAssign
+					poolTolerations = append(tcd.GPUPodSpec.Spec.Tolerations, gpuTolerations...)
+				} else if tcd.CPUPodSpec != nil {
+					//nolint:gocritic,appendAssign
+					poolTolerations = append(tcd.CPUPodSpec.Spec.Tolerations, cpuTolerations...)
+				}
+			}
 
-		// If there's no quota for our only resource pool's namespace
-		if _, ok := namespaceToQuota[namespaceOfPool]; !ok {
-			return p.summarizeClusterByNodes(ctx), nil
+			// If all of a node's taints are tolerated by a pool, that node belongs to the pool.
+			if allTaintsTolerated(node.Spec.Taints, poolTolerations) {
+				poolsToNodes[poolName] = append(poolsToNodes[poolName], node)
+			}
 		}
 	}
 
+	// Build the set of summaries for each resource pool
 	containers := p.containersPerResourcePool()
 	summaries := make(map[string]model.AgentSummary, len(p.namespaceToPoolName))
-	for namespace, poolName := range p.namespaceToPoolName {
+	for poolName, nodes := range poolsToNodes {
 		slots := model.SlotsSummary{}
-		numContainers := containers[poolName]
-		var registeredTime time.Time
-		if quota, quotaExists := namespaceToQuota[namespace]; quotaExists {
-			slots = make(map[string]model.SlotSummary)
-			registeredTime = quota.CreationTimestamp.Time
-
-			for resourceName, qty := range quota.Spec.Hard {
-				var deviceType device.Type
-				switch resourceName {
-				case k8sV1.ResourceCPU:
-					deviceType = device.CPU
-				case ResourceTypeNvidia, "limits." + ResourceTypeNvidia:
-					deviceType = device.CUDA
-				default:
-					// We only care about CPU and GPU quotas for the slots summary
-					continue
-				}
+		numContainersInPool := containers[poolName]
 
-				// Each CPU and GPU in the quota will be counted as a slot here
-				one, decQty := inf.NewDec(1, 0), qty.AsDec()
-				for i := inf.NewDec(0, 0); i.Cmp(decQty) < 0; i.Add(i, one) {
-					id := fmt.Sprintf("%s/%s/%s", poolName, string(deviceType), i.String())
-
-					var container *cproto.Container
-					// Create a number of pseudo-containers in the summary equal to the number of
-					// running containers
-					if decNumContainers := inf.NewDec(int64(numContainers),
-						0); i.Cmp(decNumContainers) < 0 {
-						container = &cproto.Container{
-							ID:    cproto.ID(id),
-							State: "RUNNING",
-						}
-					}
+		// We'll create a number of pseudo-containers in the summary equal to the number of
+		// running containers in this pool.
+		pseudoContainersAdded := 0
```

Comment:
non-blocking question: I noticed that you initialized the Go variable here w/ shorthand and later with `var int ` -- for sake of consistency & my curiosity, are there different use cases?